src/test/java/bio/terra/endpoints/CLICommandTests.java

```diff
@@ -0,0 +1,284 @@
+        List<String> cliCmdResponse = CLITestingUtils.callCLICommand(cmdArgs);
+
+        // read in the expect CLI command output from a file
+        List<String> cliCmdExpectedResponse = CLITestingUtils.readCLIExpectedOutput(expectedOutputFilename);
+
+        // log the response to stdout
+        System.out.println("cliCmdResponse: " + cliCmdResponse + "\n");
+
+        // need to replace %xyz% variables in the CLI expected response with values from the Java HTTP response
+        String id = datasetDetails.get("id").toString();
+        String createdDate = datasetDetails.get("createdDate").toString();
+        ArrayList<Object> assets =
+                (ArrayList<Object>)((Map<String, Object>)datasetDetails.get("schema")).get("assets");
+        ArrayList<Object> assetTables = (ArrayList<Object>)((Map<String, Object>)assets.get(0)).get("tables");
+        ArrayList<String> assetTableNames = new ArrayList<>();
+        for (int ctr = 0; ctr < assetTables.size(); ctr++) {
+            String tableName = ((Map<String, Object>)assetTables.get(ctr)).get("name").toString();
+            assetTableNames.add(tableName);
+        }
+
+        // check that the responses match
+        for (int ctr = 0; ctr < cliCmdResponse.size(); ctr++) {
+            String expectedLine = cliCmdExpectedResponse.get(ctr);
+            expectedLine = expectedLine.replace("%id%", id);
+            expectedLine = expectedLine.replace("%createdDate%", createdDate);
+            expectedLine = expectedLine.replace("%assetTable0%", assetTableNames.get(0));
+            expectedLine = expectedLine.replace("%assetTable1%", assetTableNames.get(1));
+
+            Assert.assertEquals(expectedLine, cliCmdResponse.get(ctr));
+        }
+    }
+
+    /**
+     * CLI command : jc dr describe
+     * Repository API : GET : enumerateDatasets
+     * @throws IOException
+     */
+    public void drDescribeTest(Map<String, Object> datasetDetails, String expectedOutputFilename)
+            throws IOException {
+        String datasetName = datasetDetails.get("name").toString();
+        System.out.println("***********************************************");
+        System.out.println("jc dr describe " + datasetName);
+
+        // call CLI command in a separate process
+        List<String> cmdArgs = new ArrayList<String>(new ArrayList<>(Arrays.asList(
+                new String[]{"dr", "describe", datasetName})));
+        List<String> cliCmdResponse = CLITestingUtils.callCLICommand(cmdArgs);
+
+        // read in the expect CLI command output from a file
+        List<String> cliCmdExpectedResponse = CLITestingUtils.readCLIExpectedOutput(expectedOutputFilename);
+
+        // log the response to stdout
+        System.out.println("cliCmdResponse: " + cliCmdResponse + "\n");
+
+        // need to replace %xyz% variables in the CLI expected response with values from the Java HTTP response
+        String id = datasetDetails.get("id").toString();
```

Comment:
Maybe DRY this out?

Reply:
Good idea. Refactored the variable swap and line by line comparison of CLI expected and actual output into separate methods in the utils class.